Models/TaskModel.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Models;
+
+class TaskModel extends DatabaseModel
+{
+    const STATUS_NEW = 3;
+    const STATUS_DONE = 2;
+    const STATUS_APPROVE = 1;
+    const STATUS_FAIL = 0;
+
+    public function createTask($data)
+    {
+        if ($_FILES['image']["error"] == UPLOAD_ERR_OK) {
+            $data['image'] = rand() . $_FILES['image']['name'];
+            move_uploaded_file($_FILES['image']['tmp_name'], 'Media/images/tasks/' . $data['image']);
+        }
+        $data['time_end'] = date("Y-m-d H:i:s", time() + 3600 * $data['time_end']);
+        $data['status'] = self::STATUS_NEW;
+        return $this->insertData('tasks', $data);
+    }
+
+    public function getAllTasks($whereCondition = null)
+    {
+        $field = [
+            't.id', 'time_created', 'u1.name AS created_by', 'u2.name AS executor', 'task',
+            'status', 'time_start', 'time_end', 'comment', 'u3.name AS approved_by', 't.image'
+        ];
+        $joinCondition = [
+            'created_by' => 'u1.id',
+            'executor' => 'u2.id',
+            'approved_by' => 'u3.id'
+        ];
+        $this->isTaskActive();
```

Comment:
do you really need to run this query every single time when page is loading this method?
it can cause a lot if redundant update table queries to dabatase. for example 5000000 people refreshed the page on this action and there will be 5000000 empty update requests to the database?